src/workerd/io/observer.h

```diff
@@ -30,6 +32,30 @@ class WebSocketObserver: public kj::Refcounted {
   virtual void receivedMessage(size_t bytes) {};
 };
 
+class MemoryCacheLockRecord {
+public:
+  explicit MemoryCacheLockRecord(SpanBuilder& span, const kj::MonotonicClock& timer)
+      : span(span),
+        timer(timer),
+        startTime(timer.now()) {}
```

Comment:
We could just use the kj::systemPreciseMonotonicClock().now() if we don't mind making a timing a measurement here.